test/yb_test.c

```diff
@@ -0,0 +1,339 @@
+
+    mu_assert("Incorrect hdr_histogram struct size", compare_int64(sizeof(hdr_histogram), 96));
+    mu_assert("Incorrect hdr_histogram total size", compare_int64(hdr_get_memory_size(h), 800));
+    mu_assert("Incorrect subbucket count", compare_int64(h->sub_bucket_count,
+        yb_default_bucket_factor));
+    mu_assert("Incorrect bucket count", compare_int64(h->bucket_count, 21));
+
+    /* 
+     * first bucket
+     */
+    hdr_record_value(h, 1);
+    hdr_record_value(h, 5);
+
+    /* 
+     * bucket becomes resolution 2, 17 will be recorded as 16
+     */
+    hdr_record_values(h, 17, 3);
+
+    /* 
+     * second-to-last-bucket
+     */
+    hdr_record_value(h, 8388607);
+
+    /* 
+     * beginning of last bucket, check these values fall into same subbucket
+     */
+    hdr_record_value(h, 8388608);
+    hdr_record_value(h, 9000000);
+
+    /* 
+     * max value and beyond max, check that 16777216 is not recorded but 16777215 is
+     */
+    hdr_record_value(h, yb_default_max);
+    hdr_record_value(h, yb_default_max + 1);
+
+    mu_assert("Incorrect first bucket value", compare_int64(h->counts[1], 1));
+    mu_assert("Incorrect first bucket value", compare_int64(h->counts[5], 1));
+
+    mu_assert("Incorrect second bucket value", compare_int64(h->counts[16], 3));
+
+    mu_assert("Incorrect penultimate bucket value", compare_int64(h->counts[167], 1));
+    mu_assert("Incorrect last bucket value", compare_int64(h->counts[168], 2));
+
+    mu_assert("Incorrect last subbucket value", compare_int64(h->counts[h->counts_len - 1], 1));
+
+    mu_assert("Incorrect subbucket width", compare_int64(get_subbucket_width(h, 123), 8));
+
+    int final_subbucket_beg = yb_default_max + 1 - get_subbucket_width(h, yb_default_max);
+
+    mu_assert("Incorrect final subbucket beginning", compare_int64(
+        lowest_equivalent_value(h, yb_default_max), final_subbucket_beg));
+    mu_assert("Incorrect final subbucket end", compare_int64(
+        highest_equivalent_value(h, yb_default_max), yb_default_max));
+
+    mu_assert("Incorrect p50 value", compare_values(hdr_value_at_percentile(h, 50), 17, 0.001));
+    mu_assert("Incorrect p90 value", compare_values(hdr_value_at_percentile(h, 90), 9437183, 0.001));
+    mu_assert("Incorrect p99 value", compare_values(hdr_value_at_percentile(h, 99), yb_default_max, 0.001));
+
+    hdr_close(h);
+
+    #endif
+
+    return 0;
+}
+
+/* 
+ * confirm we can calculate the actual histogram maximum for an arbitrary histogram max input 
+ * like 30000, using only subbucket size and bucket count
+ * we are interested in yb default config only so we skip the body when
+ * FLEXIBLE_COUNTS_ARRAY is not set up - that would be a 32 sub_bucket_count histogram
+ */
+static char* yb_derived_max_test(void)
+{
+    #if defined(FLEXIBLE_COUNTS_ARRAY) && defined(USE_SHORT_COUNT_TYPE)
+
+    int test_max = 30000;
+    hdr_histogram* dummy = (hdr_histogram*) calloc(1, sizeof(hdr_histogram));
```

Comment:
None of these allocations are checked... I know that this is just test code, but it might be nice to add an assertion to these.

```c
if (dummy == NULL) {
    mu_assert("Failed to allocate hdr_histogram", false);
}
```

Maybe just a macro would suffice?